File: backend/src/main/java/it/hva/dmci/ict/ewa/spaghetteria4/backend/models/Question.java
package it.hva.dmci.ict.ewa.spaghetteria4.backend.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;

public class Question {
    @JsonProperty
    private String name;
    @JsonProperty
    private String type;
    @JsonProperty
    private ArrayList<String> options;

    @JsonCreator
    public Question(String name, String type, ArrayList<String> options) {
        this.name = name;
        this.type = type;
        this.options = options;
    }

    public Question(String name, String type) {
        this(name,type,null);
    }

    public Question() {
    }
}

File: frontend/src/app/_models/Question.ts
export class Question {
  private _name: string;
  private _type: string;
  private _options: string[];

  constructor(name: string, type: string, options: string[]) {
    this._name = name;
    this._type = type;
    this._options = options;
  }

  get name(): string {
    return this._name;
  }

  get type(): string {
    return this._type;
  }

  get options(): string[] {
    return this._options;
  }
}
